### PR TITLE
new columns kusto upload

### PR DESCRIPTION
**Pull request by shivanispatel**
New function to upload results to the new kusto table in AzHealthChecks/AzNhcRunEvents

Schema : vmSize:string,vmId:string,vmHostname:string,physHostname:string,workflowType:string,eventTime:datetime,pass:string,errors:string,logOutput:string,jsonResult:string,uuid:string

### CODE REVIEW

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 1]
```diff
@@ -60,13 +63,155 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
...
+
+
+def get_nhc_json_formatted_result(results_file):
+    def natural_sort_key(s):
+            return [int(text) if text.isdigit() else text.lower() for text in re.split(r'(\d+)', s)]
+
+    # check if CPU or GPU 
+    stream_Copy_cmd = f"cat {results_file} | grep -o 'stream_Copy: .*'"
+    stream_Copy_str = run_command(stream_Copy_cmd)
+    n = "CPU" if stream_Copy_str else "GPU"
```
Comment (rafsalas19) [inline distributed_nhc/export_nhc_result_to_kusto.py:79]: Can we give this variable "n" a better name

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 2]
```diff
@@ -60,13 +63,155 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
...
+            elif key.startswith("H2D"):
+                result["GPU_BW_HTD"][key] = str(value.strip())
+            elif key.startswith("D2H"):
+                result["GPU_BW_DTH"][key] = str(value.strip())
+            elif key.startswith("P2P"):
+                result["GPU_BW_P2P"][key] = str(value.strip())
+            elif key.startswith("nccl_all_red"):
+                result["NCCL_ALL_REDUCE"] = str(value.strip())
+            elif key.startswith("nccl_all_red_lb"):
+                result["NCCL_ALL_REDUCE_LOOP_BACK"] = str(value.strip())
```
Comment (rafsalas19) [inline distributed_nhc/export_nhc_result_to_kusto.py:156]: Is there a reason we don't explicitly return results dictionary in this function?

**Comment by rafsalas19** [comment]
Hey Shivani, can you squash the commits into 2-3 meaningful commits.

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 3]
```diff
@@ -60,13 +63,155 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
...
+        P2P_GPU_cmd = f"cat {results_file} | grep -o 'P2P_GPU_[0-7]_[0-7]: .*'"
+        P2P_GPU_str = run_command(P2P_GPU_cmd)
+
+        nccl_all_red_cmd = f"cat {results_file} | grep -o 'nccl_all_red: .*'"
+        nccl_all_red_str = run_command(nccl_all_red_cmd)
+
+        nccl_all_red_lb_cmd = f"cat {results_file} | grep -o 'nccl_all_red_lb: .*'"
+        nccl_all_red_lb_str = run_command(nccl_all_red_lb_cmd)
+
+        data_string = "\n".join([ib_write_lb_mlx5_ib_str, H2D_GPU_str, D2H_GPU_str, P2P_GPU_str, nccl_all_red_str, nccl_all_red_lb_str])
```
Comment (rafsalas19) [inline distributed_nhc/export_nhc_result_to_kusto.py:138]: It looks like I am failing here. The issue is that "ib_write_lb_mlx5_ib_str" seems to be a list. In the CPU section we converted it to a string. We need to do the same for the GPU section. i.e.  ib_write_lb_mlx5_ib_str = '\n'.join(ib_write_lb_mlx5_ib_str) # convert to string

File: distributed_nhc/export_nhc_result_to_kusto.py (hunk 2 above)
Comment (rafsalas19) [reply distributed_nhc/export_nhc_result_to_kusto.py:156]: When I tested this I needed to add the return result to make this part work

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 4]
```diff
@@ -60,13 +63,155 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
...
+        uuid = "-" + uuid
+    full_uuid = f"nhc-{ts}{uuid}"
+
+    vmSize_bash_cmd = "echo $( curl -H Metadata:true --max-time 10 -s \"http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2021-01-01&format=text\") | tr '[:upper:]' '[:lower:]' "
+    vmSize = run_command(vmSize_bash_cmd)
+
+    vmId_bash_cmd = "curl  -H Metadata:true --max-time 10 -s \"http://169.254.169.254/metadata/instance/compute/vmId?api-version=2021-02-01&format=text\""
+    vmId = run_command(vmId_bash_cmd)
+
+    vmName_bash_cmd = "timeout 60 sudo /opt/azurehpc/tools/kvp_client | grep \" HostName; \" | grep -o 'Value: [^;]*' | cut -d ' ' -f 2" # keep the spaces, else it will also output the results for 'PhysicalHostName'
```
Comment (yakovdyadkin) [inline distributed_nhc/export_nhc_result_to_kusto.py:177]: I think vmName should just be the VM's hostname, `hostname` cmd

File: distributed_nhc/export_nhc_result_to_kusto.py [hunk 5]
```diff
@@ -45,7 +48,7 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
     if job_name == "pssh":
         job_name = f"{job_name}-{ts_str}"
 
-    with open(health_file, 'r') as f:
+    with open(health_file, 'r') as f: # should this be debug file?
```
Comment (yakovdyadkin) [inline distributed_nhc/export_nhc_result_to_kusto.py:51]: I think you're correct it should be `debug_file`

**Comment by shivanispatel** [comment]
Addressing comments; moving all changes to a new PR for cleaner commit history

File: distributed_nhc/export_nhc_result_to_kusto.py (hunk 2 above)
Comment (shivanispatel) [reply distributed_nhc/export_nhc_result_to_kusto.py:156]: Added the return statement

**Pull request closed without merging**